yt/frontends/enzo_p/io.py

```diff
@@ -82,34 +98,43 @@ def _read_particle_fields(self, chunks, ptf, selector):
         for chunk in chunks: # These should be organized by grid filename
             f = None
             for g in chunk.objs:
-                if g.filename is None: continue
+                if g.filename is None:
+                    continue
                 if f is None:
                     f = h5py.File(g.filename, "r")
-                nap = sum(g.NumberOfActiveParticles.values())
-                if g.NumberOfParticles == 0 and nap == 0:
+                if g.particle_count is None:
+                    fnstr = "%s/%s" % \
+                      (g.block_name, self._sep.join(["particle", "%s", "%s"]))
+                    g.particle_count = \
+                      dict((ptype, f.get(fnstr %
+                            (ptype, self.sample_pfields[ptype])).size)
+                            for ptype in self.sample_pfields)
+                    g.total_particles = sum(g.particle_count.values())
+                if g.total_particles == 0:
                     continue
-                ds = f.get(g.block_name)
+                group = f.get(g.block_name)
                 for ptype, field_list in sorted(ptf.items()):
-                    if ptype != "io":
-                        if g.NumberOfActiveParticles[ptype] == 0: continue
-                        pds = ds.get("Particles/%s" % ptype)
-                    else:
-                        pds = ds
-                    pn = _particle_position_names.get(ptype,
-                            r"particle_position_%s")
-                    x, y, z = (np.asarray(pds.get(pn % ax).value, dtype="=f8")
-                               for ax in 'xyz')
+                    pn = self._sep.join(
+                        ["particle", ptype, "%s"])
+                    if g.particle_count[ptype] == 0:
+                        continue
+                    coords = \
+                      tuple(np.asarray(group.get(pn % ax).value, dtype="=f8")
+                            for ax in 'xyz'[:self.ds.dimensionality])
+                    for i in range(self.ds.dimensionality, 3):
+                        coords += (self.ds.domain_center[i].d *
```

Comment:
At present, it *is* possible for this to be changed into different units.  It *might* be safer (he said, not taking this advice himself) to set up a view with something like `DC = self.ds.domain_center.in_units("code_length").d` and use that.

Reply:
yeah, for example if we change the units of `domain_center` in the future this would break.

Reply:
Ok, will do.